### PR TITLE
Code cleanup from code review with Tim/Abhipsa

### PR DESCRIPTION
**Pull request by drwill-ms**

### CODE REVIEW

File: iothub/device/src/Authentication/AuthenticationMethodFactory.cs [hunk 1]
```diff
@@ -1,49 +0,0 @@
...
-using System;
-
-namespace Microsoft.Azure.Devices.Client
-{
-    /// <summary>
-    /// Creates an instance of an implementation of <c>IAuthenticationMethod</c> based on known authentication parameters.
-    /// </summary>
-    internal sealed class AuthenticationMethodFactory
-    {
-        internal static IAuthenticationMethod GetAuthenticationMethodFromConnectionString(IotHubConnectionString iotHubConnectionString)
```
Comment (drwill-ms) [inline iothub/device/src/Authentication/AuthenticationMethodFactory.cs:13]: Moved to the one class that used it.

File: iothub/device/src/Authentication/HsmAuthentication/GeneratedCode/HttpHsmClient.cs [hunk 2]
```diff
@@ -158,411 +157,10 @@ public async Task<SignResponse> SignAsync(string apiVersion, string name, string
         }
 
         /// <param name="apiVersion">The version of the API.</param>
-        /// <param name="name">The name of the module on whose behalf the plaintext will be encrypted. (urlencoded)</param>
-        /// <param name="genid">The generation identifier for the module as generated by IoT hub.</param>
-        /// <param name="payload">The data to be encrypted.</param>
-        /// <returns>OK</returns>
-        /// <exception cref="SwaggerException">A server side error occurred.</exception>
-        /// <param name="cancellationToken">A cancellation token that can be used by other objects or threads to receive notice of cancellation.</param>
-        public async Task<EncryptResponse> EncryptAsync(string apiVersion, string name, string genid, EncryptRequest payload, CancellationToken cancellationToken = default)
```
Comment (drwill-ms) [inline iothub/device/src/Authentication/HsmAuthentication/GeneratedCode/HttpHsmClient.cs:167]: Removed methods with 0 references.

File: iothub/device/src/Authentication/HsmAuthentication/HttpClientHelper.cs [hunk 3]
```diff
@@ -27,7 +28,7 @@ public static HttpClient GetHttpClient(Uri providerUri)
 
             if (providerUri.Scheme.Equals(UnixScheme, StringComparison.OrdinalIgnoreCase))
             {
-                return new HttpClient(new HttpUdsMessageHandler(providerUri));
+                return new HttpClient(new HttpUdsMessageHandler(providerUri), true);
```
Comment (drwill-ms) [inline iothub/device/src/Authentication/HsmAuthentication/HttpClientHelper.cs:31]: Avoid this memory leak.

Comment (drwill-ms) [reply iothub/device/src/Authentication/HsmAuthentication/HttpClientHelper.cs:31]: https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpclient.-ctor?view=net-7.0#system-net-http-httpclient-ctor(system-net-http-httpmessagehandler-system-boolean)

File: iothub/device/src/Authentication/HsmAuthentication/HttpHsmComunicationException.cs [hunk 4]
```diff
@@ -4,20 +4,22 @@
...
+    internal class HttpHsmComunicationException : Exception
     {
         /// <summary>
         /// Creates an instance of this class with the supplied error message and status code.
         /// </summary>
         /// <param name="message">The message that describes the error.</param>
         /// <param name="statusCode">Status code of the communication failure.</param>
-        public HttpHsmComunicationException(string message, int statusCode) : base($"{message}, StatusCode: {statusCode}")
+        /// <param name="innerException">The inner exception, if any.</param>
+        public HttpHsmComunicationException(string message, int statusCode, Exception innerException = default)
```
Comment (drwill-ms) [inline iothub/device/src/Authentication/HsmAuthentication/HttpHsmComunicationException.cs:21]: maintain the inner exception

File: iothub/device/src/Authentication/HsmAuthentication/Transport/UnixDomainSocketEndPoint.cs [hunk 5]
```diff
@@ -12,21 +12,13 @@ namespace Microsoft.Azure.Devices.Client.HsmAuthentication.Transport
 {
     internal sealed class UnixDomainSocketEndPoint : EndPoint
     {
-        private const AddressFamily EndPointAddressFamily = AddressFamily.Unix;
```
Comment (drwill-ms) [inline iothub/device/src/Authentication/HsmAuthentication/Transport/UnixDomainSocketEndPoint.cs:15]: Remove unreferenced class members

**Review by drwill-ms — COMMENTED** [review]
Mostly just code style changes and removing dead code.

**Review by brycewang-microsoft — APPROVED** [review]

**Review by abhipsaMisra — COMMENTED** [review]
LGTM

**Pull request merged**